src/h5web/vis-packs/core/utils.test.ts

```diff
@@ -97,9 +97,23 @@ describe('Shared visualization utilities', () => {
       expect(extendedDomain).toEqual([-50, 150]);
     });
 
-    it('should extend domain by factor for use with log scale', () => {
-      const extendedDomain = extendDomain([1, 100], 0.25, true);
-      expect(extendedDomain).toEqual([0.8, 125]);
+    it('should return a non-empty domain when given an empty domain', () => {
+      const extendedDomain = extendDomain([1, 1], 0.5);
+      expect(extendedDomain).toEqual([0, 2]);
+    });
+
+    it('should extend domain for use with log scale', () => {
+      const [min, max] = [10, 100];
+      const [extMin, extMax] = extendDomain([10, 100], 0.25, ScaleType.Log);
+      expect(extMin < min).toBe(true);
+      expect(extMax > max).toBe(true);
```

Comment:
It is now hard to check for the exact value...

Reply:
Isn't there a combination of domain and extension ratio that could result in a predictable extended domain?

Reply:
The extended bound for the log depends on `10**extendFactor` which is almost never an integer for `0 < extendFactor <1`. 

Mind, I could use`toBeCloseTo` to do the test with floating numbers. But to deduce the extended value, I was kinda writing the formula of `scale.invert` which makes the test eats its own tail...

I could perhaps use `extendFactor = 1` for the test :shrug:

Reply:
Yeah, one test with `1`, one test with `0` and one in the middle where you check that the value `isBetween` two values.

Reply:
Maybe there are some edges cases to check too, like when the domain has a bound equal to `0` in linear or symlog.

Reply:
And maybe add checks in the empty domain test with the other scales? This will document the behaviour more precisely, and will avoid any future regressions.